compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/DecomposeWinogradPass.cpp

```diff
@@ -227,32 +223,38 @@ struct DecomposeWinogradInputTransform
 
     auto inputExtractSliceOp =
         dynamicSlice.getDefiningOp<tensor::ExtractSliceOp>();
-    SmallVector<OpFoldResult> mixedSizes = inputExtractSliceOp.getMixedSizes();
     // Harcoding input rank as 4 here - since we'd be getting a tiled version
     // with rank 2. We are always expected to either have a rank 4 version of
     // this op, or rank 2 (tiled). And at this point in the flow, it is
     // guaranteed to be a rank 2 version of the op as ensured by the assertion
-    // above. Copy input slice into zeroed padded scratch space
-    SmallVector<OpFoldResult> offsets(2, zero);
-    SmallVector<OpFoldResult> sizes;
-    SmallVector<OpFoldResult> strides(2, one);
-    for (int i = 0; i < 4; i++) {
-      if (imageDimsSet.contains(i)) {
-        sizes.push_back(mixedSizes[i]);
-      }
+    // above. Pad the input slice.
+
+    SmallVector<OpFoldResult> mixedSizes = inputExtractSliceOp.getMixedSizes();
+    SmallVector<OpFoldResult> dynamicSliceSizes;
+    for (auto idx : transformOp.getImageDimensions()) {
+      dynamicSliceSizes.push_back(mixedSizes[idx]);
     }
-    linalg::FillOp fillOp = rewriter.create<linalg::FillOp>(
-        loc, ValueRange{zeroF32}, ValueRange{scratch});
-    Value inputSlice = rewriter.create<tensor::InsertSliceOp>(
-        loc, dynamicSlice, fillOp.result(), offsets, sizes, strides);
+    SmallVector<OpFoldResult> padLow(inputTileSquare.size(),
+                                     rewriter.getIndexAttr(0));
+    SmallVector<OpFoldResult> padHigh;
+    for (auto [idx, size] : llvm::enumerate(dynamicSliceSizes)) {
```

Comment:
actually instead of creating a vector for `dynamicSliceSizes`, can you just use `transformOp.getImageDimensions()` directly?